packages/core/src/components/fields/SchemaField.tsx

```diff
@@ -249,6 +249,12 @@ function SchemaFieldRender<T, F>(props: FieldProps<T, F>) {
   if (!hideError && errors && errors.length > 0) {
     classNames.push("field-error has-error has-danger");
   }
+  if (uiSchema?.classNames) {
+    console.warn(
+      "WARNING: 'uiSchema.classNames' is deprecated and will be removed in the next major release; Use 'ui:classNames' instead."
```

Comment:
maybe we can discuss on Friday, but I'm not quite sure if we should ever remove this at all. uiSchema is rjsf's own creation, it's not like we're trying to conform to some spec like JSON Schema.

I don't see the point in ever removing classNames -- we can of course change all examples to use ui:classNames, but I see no harm in continuing to support it indefinitely.

Reply:
@epicfaace I was adding a consistency to the `uiSchema` for options. `classNames` was literally the only `uiSchema` option that did not come with the `ui:` prefix so I was making it consistent with all the others.

Reply:
Right, consistency is good, and is a good reason to make "ui:classNames" the "official" way to do it. But I don't think consistency dictates that we also have to remove support for "classNames".

Reply:
I understand... I also wonder if you didn't have your uiSchema files in a DB making it harder to just globally change them, would you feel so strongly about it keeping this inconsistency?

Reply:
Got it... Well it is deprecated for the moment, with a warning it may be removed in a future release